components/governance/org.wso2.carbon.governance.registry.extensions/src/org/wso2/carbon/governance/registry/extensions/executors/apistore/RestServiceToAPIExecutor.java

```diff
@@ -84,22 +86,30 @@ public void init(Map parameterMap) {
         // Retrieves the secured password as follows
         secretResolver.init(GovernanceRegistryExtensionsComponent.getSecretCallbackHandlerService()
                                                                  .getSecretCallbackHandler());
-        if (secretResolver != null && secretResolver.isInitialized()) {
-            apimUsername = secretResolver.resolve("apim.username");
-        }
-        if (secretResolver != null && secretResolver.isInitialized()) {
-            apimPassword = secretResolver.resolve("apim.password");
-        }
-
         this.parameterMap = parameterMap;
         if (parameterMap.get(APIM_ENDPOINT) != null) {
             apimEndpoint = parameterMap.get(APIM_ENDPOINT).toString();
         }
-        if (parameterMap.get(APIM_USERNAME) != null) {
-            apimUsername = parameterMap.get(APIM_USERNAME).toString();
+        if (parameterMap.get(ExecutorConstants.APIM_USERNAME) != null) {
```

Comment:
Since using the same code block in several classes, may be better to move it to a seperate class and invoke. Otherwose even in future changes need to be done in multiple,places.